sdk/samples/iot/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+
+if (TRANSPORT_PAHO)
+
+cmake_minimum_required (VERSION 3.10)
+
+project (az_iot_samples_common LANGUAGES C)
+
+set(CMAKE_C_STANDARD 99)
+
+find_package(eclipse-paho-mqtt-c CONFIG)
+if(NOT eclipse-paho-mqtt-c_FOUND)
+  find_package(eclipse-paho-mqtt-c REQUIRED)
+endif()
+
+find_package(OpenSSL CONFIG)
+if(NOT OpenSSL_FOUND)
+  find_package(OpenSSL REQUIRED)
+endif()
+
+# Azure IoT Samples Library
+add_library (az_iot_samples_common
+  ${CMAKE_CURRENT_LIST_DIR}/src/sample_sas_utility.c
+)
+
+target_include_directories (az_iot_samples_common
+  PUBLIC
+  ${CMAKE_CURRENT_LIST_DIR}/inc
+)
+
+# Internal deps
+target_link_libraries(az_iot_samples_common
+  PUBLIC
+    az::iot::hub
+    az::iot::provisioning
+)
+
+# External deps
+target_link_libraries(az_iot_samples_common
+  PUBLIC
+    eclipse-paho-mqtt-c::paho-mqtt3cs-static
+    OpenSSL::SSL
+    OpenSSL::Crypto
+)
+
+add_library (az::iot::samples::common ALIAS az_iot_samples_common)
+
+add_subdirectory(hub)
+add_subdirectory(provisioning)
+
+endif() # TRANSPORT_PAHO
+
```

Comment:
nit: remove the extra new line

Reply:
extra line is standard to include. it is even required for compilation at times in MacOS

Reply:
I know we want **one** extra newline character at the end of the last line in the file, even GitHub shows an error icon for that if we don't, but it looks like in this PR we are adding an extra line on top of that. Other `CmakeLists.txt` files in the repo don't show this extra line number.

Compare these:
https://github.com/Azure/azure-sdk-for-c/blob/099446822c992b911f83443d28e92f092bb16011/CMakeLists.txt#L116
https://github.com/Azure/azure-sdk-for-c/blob/099446822c992b911f83443d28e92f092bb16011/sdk/src/azure/iot/CMakeLists.txt#L78
https://github.com/Azure/azure-sdk-for-c/blob/099446822c992b911f83443d28e92f092bb16011/sdk/tests/iot/hub/CMakeLists.txt#L27

To this one:
https://github.com/Azure/azure-sdk-for-c/blob/099446822c992b911f83443d28e92f092bb16011/sdk/samples/iot/CMakeLists.txt#L52-L53

I am betting we can remove that and it will compile fine on MacOS (given CI passes with the other files) and GitHub won't show the error icon either:
https://0.readthedocs.io/en/latest/git.html#check-there-is-a-new-line-at-end-of-file
![image](https://user-images.githubusercontent.com/6527137/87613698-e9ff5000-c6c2-11ea-8a58-d447c598345e.png)